drivers/sensor/iis2mdc/iis2mdc_trigger.c

```diff
@@ -107,13 +107,12 @@ static void iis2mdc_work_cb(struct k_work *work)
 int iis2mdc_init_interrupt(const struct device *dev)
 {
 	struct iis2mdc_data *iis2mdc = dev->data;
-	const struct iis2mdc_config *const config = dev->config;
+	const struct iis2mdc_dev_config *const config = dev->config;
 
 	/* setup data ready gpio interrupt */
-	iis2mdc->gpio = device_get_binding(config->drdy_port);
+	iis2mdc->gpio = config->gpio_drdy.port;
 	if (iis2mdc->gpio == NULL) {
-		LOG_DBG("Cannot get pointer to %s device",
-			    config->drdy_port);
+		LOG_ERR("Cannot get pointer to drdy_gpio device");
```

Comment:
You also need to update this to use device_is_ready() or device_usable_check() instead. The device pointer can be non-NULL but point to a device which is not initialized.

Reply:
I'll use device_is_ready()

Reply:
> I'll use device_is_ready()

You need to do this for all the devices you depend on. That includes the bus too.

Reply:
Rad yesterday evening, but it was too late and my mind was totally blank... :D 
So, I think I'll better move this che and the bus check in the main driver init. I'll check all the devices there (I think I have to check gpio device for drdy, and device for bus. I don't think something else is left).

Reply:
Mmmmmh, and SPI cs gpio_dev too. So, maybe better to keep the checks in different routines! Let me know if I'm missing something again... :(